public/apps/account/account-nav-button.tsx

```diff
@@ -186,3 +186,24 @@ export function AccountNavButton(props: {
     </EuiHeaderSectionItemButton>
   );
 }
+
+export function reloadAfterTenantSwitch(): void {
+  // the below portion is to clear URLs starting with 'lastUrl'
+  // when switching tenants, the last URLs will be from the old tenancy therefore we need to remove these from sessionStorage.
+  const lastUrls = [];
+  const { sessionStorage } = window;
+
+  for (let i = 0; i < sessionStorage.length; i++) {
+    const key = sessionStorage.key(i);
+    if (key?.startsWith('lastUrl')) {
+      lastUrls.push(key);
+    }
+  }
+  for (let i = 0; i < lastUrls.length; i++) {
+    sessionStorage.removeItem(lastUrls[i]);
```

Comment:
I'm curious, why is removing `lastUrl:` entries from session storage required here? When testing locally, I see that `lastUrl:` entries are kept when using the tenant switcher to switch tenants in the middle of a session. Why is it required when loading from a URL?

Reply:
Hi @cwperks, this body of work was to resolve the issue we were seeing where, when a URL was copied, then the user pastes that URL and attempts to change the tenant, they could not. It would revert back to the old URL. Keeping that in mind if the tenant has been changed, then the last stored URLs could be from pages/dashboards, no longer accessible to the changed tenant causing an error.

Reply:
Oh I see. I can see the values of `lastUrl` in the session storage when testing locally. Is it possible to pull this constant from OSD Core or is it not available?